types/test.ts

```diff
@@ -189,29 +189,32 @@ function testParametricSelector() {
     (state: {testStringArray: string[]}) => state.testStringArray,
     (foo1: string, foo2: number, foo3: boolean, foo4: string, foo5: string, foo6: string, foo7: string, foo8: string, foo9: string[]) => {
       return {foo1, foo2, foo3, foo4, foo5, foo6, foo7, foo8, foo9};
-    });
+    },
+  );
+
+  heterogeneousSelector({testString: 'foo', testBoolean: true, testNumber: 2})
 
   const selector = createSelector(
     (state: State) => state.foo,
-    (state: never, props: Props) => props.bar,
+    (state: State, props: Props) => props.bar,
     (foo, bar) => ({foo, bar}),
   );
 
-  // typings:expect-error
+  // $ExpectError
   selector({foo: 'fizz'});
-  // typings:expect-error
+  // $ExpectError
   selector({foo: 'fizz'}, {bar: 'baz'});
 
   const ret = selector({foo: 'fizz'}, {bar: 42});
   const foo: string = ret.foo;
   const bar: number = ret.bar;
 
   const selector2 = createSelector(
-    (state) => state.foo,
-    (state) => state.foo,
-    (state) => state.foo,
-    (state) => state.foo,
-    (state) => state.foo,
+    (state: any) => state.foo,
+    (state: any) => state.foo,
+    (state: any) => state.foo,
+    (state: any) => state.foo,
+    (state: any) => state.foo,
```

Comment:
Non-explicit `any` is generally disallowed, we will enforce this via our type definitions. This can be disabled with a `tsconfig.json` change.